### PR TITLE
test(button): added unit tests

### PR DESCRIPTION
**Pull request by alfredskeden**
**Describe pull-request**  
Followed the test steps in ticket/excel to test.

**Solving issue**  
Fixes: [CDEP-2961](https://tegel.atlassian.net/browse/CDEP-2961)

**How to test**  
1. Navigate to packages/core.
2. Check the code for tests in button component.
3.  Make sure docker is running.
4. Then run `npm run test` in core to run the tests. Make sure they run.

**Additional context** 
I'm not sure how we would like to check for colors on the button, but this way I found works for me.
I'm 100% sure I missed some tests and that some tests dont make sense. But they all run and will break if something changes on the button that is changing.
We are still in exploration for good testing.

[CDEP-2961]: https://tegel.atlassian.net/browse/CDEP-2961?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mJarsater** [comment]
🔥🚀

File: packages/core/src/components/button/test/icon/index.html
Comment (theJohnnyMe) [inline packages/core/src/components/button/test/icon/index.html:18]: Please double check this with @TatyanaBumazhkina and [test case spreadsheet.](https://scaniaazureservices-my.sharepoint.com/:x:/r/personal/tatyana_bumazhkina_scania_com/Documents/Unit%20test%20scenarios.xlsx?d=wca48bf6d92cf4e8c8f456337e2002ec4&csf=1&web=1&e=rl48tp)

Scenario no.10 I understood as "Only icon button with no text" on it so please, double check it :)

**Review by theJohnnyMe — COMMENTED** [review]
Great work man, so many test cases you did.
There is one doubt for "Only icon" story, otherwise looks good.

File: packages/core/src/components/button/test/icon/index.html
Comment (alfredskeden) [reply packages/core/src/components/button/test/icon/index.html:18]: Aaah yes ofc :) Thats is what that meant. Im so new for these components and I never use a button with only a component but mostly with combination of texts. I will check the testa for only a icon and look for documentation on how to do that. Thanks for checking.

Comment (alfredskeden) [reply packages/core/src/components/button/test/icon/index.html]: @theJohnnyMe updated, And also found that lg and sm should have fixed height and width for buttons with an icon so I added tests for that. atleast for lg size.

**Review by theJohnnyMe — APPROVED** [review]
Looks good to me.
Please, before merging, edit the PR title so once it is squashed and merged it is in small caps: _test(button): adding unit tests_

File: packages/core/src/components/button/test/basic/button.e2e.ts
Comment (timrombergjakobsson) [inline packages/core/src/components/button/test/basic/button.e2e.ts:4]: Nice! Is this a common practice, or more a personal style :)?

Comment (alfredskeden) [reply packages/core/src/components/button/test/basic/button.e2e.ts:4]: Totally personal style :) 
Like for now with dropdown I do this
```
const componentTestPath = 'src/components/dropdown/test/default/index.html';
```
and then 
```
await page.goto(componentTestPath);
```

Its just so I don't have to update the `await page.goto(componentTestPath);` every time I copy a testcase :D

Comment (alfredskeden) [reply packages/core/src/components/button/test/basic/button.e2e.ts:4]: We would still need to decide on what approach to take :) Or we just do whatever Martin proposed. I think it all works. I also tried `beforeEach` test but that did not work with spying on clicks and so on.

Comment (alfredskeden) [reply packages/core/src/components/button/test/basic/button.e2e.ts]: Changed to componentTestPath becouse I think its abit more clear on what is happening and less text. :)

Comment (timrombergjakobsson) [reply packages/core/src/components/button/test/basic/button.e2e.ts]: I see, I think we in the team should decide on what approach we want to take. I don't know what Martin proposed. I think we are just now starting out to test so maybe we should have that talk soon. So we create some common ways of doing things. We have the "How we test" in [IXIB-book](https://scaniaazureservices-my.sharepoint.com/personal/nikola_johnny_mirkovic_scania_com/_layouts/15/Doc.aspx?sourcedoc={a2c3d5d9-d0dc-4cb9-bee5-5e8a81cedb1b}&action=edit&wd=target%28Testing.one%7C5706639b-95cd-40d8-95f6-f45fe9e25134%2FHow%20we%20write%20unit%20tests%7C7c94762d-daac-4f92-a760-89a1f3b8ddd7%2F%29&wdorigin=NavigationUrl). And then, some things will always be a personal style and that is totally fine I think.

Comment (alfredskeden) [reply packages/core/src/components/button/test/basic/button.e2e.ts]: @timrombergjakobsson Yes it looks like I'm following the IXIB-book in this PR.
It mentions nothing about how we would like to specify playwright page.goto path so I just did what made most sense to me but we might need to spend an hour to talk about it?

File: packages/core/src/components/button/test/ghost/button.e2e.ts
Comment (nathalielindqvist) [inline packages/core/src/components/button/test/ghost/button.e2e.ts:71]: Should this description say it's looking for classes ghost and xs?

File: packages/core/src/components/button/test/secondary/button.e2e.ts
Comment (nathalielindqvist) [inline packages/core/src/components/button/test/secondary/button.e2e.ts:71]: Should this say it's looking for classes secondary and md?

Comment (alfredskeden) [reply packages/core/src/components/button/test/secondary/button.e2e.ts]: [687ce4d](https://github.com/scania-digital-design-system/tegel/pull/433/commits/687ce4dc6710ba31502b8d5f53cfff6c64ffdc44) Updated

File: packages/core/src/components/button/test/danger/button.e2e.ts
Comment (alfredskeden) [reply packages/core/src/components/button/test/danger/button.e2e.ts]: [687ce4d](https://github.com/scania-digital-design-system/tegel/pull/433/commits/687ce4dc6710ba31502b8d5f53cfff6c64ffdc44)

**Review by nathalielindqvist — APPROVED** [review]
LGTM!

**Pull request merged**
